sky/skypilot_config.py

```diff
@@ -78,10 +80,56 @@
 # Path to the local config file.
 CONFIG_PATH = '~/.sky/config.yaml'
 
-logger = sky_logging.init_logger(__name__)
+
+class NestedConfig(Dict[str, Any]):
+    """A nested dictionary that allows for setting and getting values."""
+
+    def get_nested(self,
+                   keys: Tuple[str, ...],
+                   default_value: Any,
+                   override_configs: Optional[Dict[str, Any]] = None) -> Any:
+        """Gets a nested key.
+
+        If any key is not found, or any intermediate key does not point to a
+        dict value, returns 'default_value'.
+
+        Args:
+            keys: A tuple of strings representing the nested keys.
+            default_value: The default value to return if the key is not found.
+            override_configs: A dict of override configs with the same schema as
+                the config file, but only containing the keys to override.
+
+        Returns:
+            The value of the nested key, or 'default_value' if not found.
+        """
+        config = copy.deepcopy(self)
+        if override_configs is not None:
+            config = _recursive_update(config, override_configs)
+        return _get_nested(config, keys, default_value)
+
+    def set_nested(self, keys: Tuple[str, ...], value: Any) -> None:
+        """Returns a deep-copied config with the nested key set to value.
```

Comment:
Docstr is inaccurate? This changes it in-place.

Reply:
Updated. Thanks!